pkg/resources/user/userHelper.go

```diff
@@ -232,3 +208,38 @@ func GetIdentitiesByProviderName(ctx context.Context, serverClient k8sclient.Cli
 
 	return identitiesByProvider, nil
 }
+
+func GetMultiTenantUsers(ctx context.Context, serverClient k8sclient.Client) (users []MultiTenantUser, err error) {
+	identities, err := GetIdentitiesByProviderName(ctx, serverClient, "testing-idp")
+	if err != nil {
+		return nil, fmt.Errorf("Error getting identity list for multi tenants")
+	}
+	for _, identity := range identities.Items {
+
+		var email = ""
+		if identity.Extra["email"] != "" {
+			email = identity.Extra["email"]
+		} else {
+			email = identity.User.Name + "@rhmi.io"
+		}
+
+		users = append(users, MultiTenantUser{
+			Username:   identity.User.Name,
+			TenantName: SanitiseTenantUserName(identity.User.Name),
+			Email:      email,
+			UID:        string(identity.User.UID),
+		})
+	}
+	return users, nil
+}
+
+func SanitiseTenantUserName(username string) string {
+	// Regex for only alphanumeric values
+	reg, _ := regexp.Compile("[^a-zA-Z0-9]+")
+
+	// Replace all non-alphanumeric values with the replacement character
+	processedString := reg.ReplaceAllString(strings.ToLower(username), invalidCharacterReplacement)
```

Comment:
If I am following this correctly this set every character to lower case and will replace any non-alphanumeric with a hyphen ? which should fix the issue with realm creation 
I will attempt to verifiy

Reply:
yep, thats correct